### PR TITLE
Add dataset verification checks

### LINKED ISSUE #196: Verification of Dataset Versioning
To support with this PR: https://github.com/BHoM/Versioning_Toolkit/pull/194/files

Once that PR is merged, we want to include verification checks for Dataset Library paths from version to version.

This will require a new `json` file added to the datasets of each version and an verification that any paths are still valid or have been upgraded and reporting errors back to PRs where that might not be the case.

FYI: @IsakNaslundBh

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
Fixes #196

### CODE REVIEW

File: .ci/code/Versioning_Test/Versioning_Test.csproj [hunk 1]
```diff
@@ -41,6 +41,9 @@
       <Private>False</Private>
       <SpecificVersion>False</SpecificVersion>
     </Reference>
+    <Reference Include="Library_Engine">
+      <HintPath>..\..\..\..\..\..\..\..\..\ProgramData\BHoM\Assemblies\Library_Engine.dll</HintPath>
```
Comment (FraserGreenroyd) [inline .ci/code/Versioning_Test/Versioning_Test.csproj:45]:
```suggestion
      <HintPath>$(ProgramData)\BHoM\Assemblies\Library_Engine.dll</HintPath>
      <Private>False</Private>
      <SpecificVersion>False</SpecificVersion>
```

**Comment by FraserGreenroyd** [comment]
@BHoMBot check project-compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check core

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Review by albinber — APPROVED** [review]
Ran versioning tests locally. Versioning check is happy.

**Pull request merged**